src/parser/jobs/sam/modules/AoeChecker.js

```diff
@@ -0,0 +1,136 @@
+import {t} from '@lingui/macro'
+import {Trans, Plural} from '@lingui/react'
+import React from 'react'
+
+import ACTIONS from 'data/ACTIONS'
+import Module from 'parser/core/Module'
+import {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+//All sets contain AoE moves to be checked, grouped by type (GCD, SEN, KENKI)
+const AOE_GCDS = new Set([ACTIONS.FUGA.id, ACTIONS.OKA.id, ACTIONS.MANGETSU.id])
+const AOE_SENS = new Set([ACTIONS.TENKA_GOKEN.id, ACTIONS.KAESHI_GOKEN.id])
+const AOE_KENKI = new Set([ACTIONS.HISSATSU_KYUTEN.id, ACTIONS.HISSATSU_GUREN.id])
+
+//These sets are based on break Points, and not type of resource used
+
+const GAIN_AT_3 = new Set([ACTIONS.FUGA.id, ACTIONS.OKA.id, ACTIONS.MANGETSU.id, ACTIONS.HISSATSU_KYUTEN.id])
+const GAIN_AT_2 = new Set([ACTIONS.HISSATSU_GUREN.id, ACTIONS.TENKA_GOKEN.id, ACTIONS.KAESHI_GOKEN.id])
+
+export default class AoeChecker extends Module {
+	static handle = 'aoechecker'
+	static title = t('sam.aoechecker.title')`AoeChecker`
+	static dependencies = [
+		'suggestions',
+	]
+
+	_badAoeGCDs = 0 //amount of aoe GCDs that didn't hit their break point for a gain
+	_badSenCasts = 0 //amount of Goken/Tsubame Goken that didn't hit the break point for damage gain
+	_badKenkiCasts = 0 //amount of Kenki Casts that didn't break the damage gain threshold
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('aoedamage', {by: 'player'}, this._onAoe)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onAoe(event) {
+
+		const action = event.ability.guid
+
+		//Step 1: Find break point
+
+		if (GAIN_AT_2.has(action)) {
+
+			//Step 2: Check break point
+
+			if (event.hits.length < 2) {
+				//Step 3: Check type of resource used and increment
+
+				if (AOE_GCDS.has(action)) {
+
+					this._badAoeGCDs++
+				} else if (AOE_SENS.has(action)) {
+
+					this._badSenCasts++
+				} else if (AOE_KENKI.has(action)) {
+
+					this._badKenkiCasts++
+				}
+			}
+
+		} else if (GAIN_AT_3.has(action)) {
+
+			//Step 2: Check break point
+
+			if (event.hits.length < 3) {
```

Comment:
Since lint complains this is a magic number, maybe convert to big const so that everyone doesn't have to see the warning pop up everytime :p